corehq/apps/linked_domain/tests/test_linked_apps.py

```diff
@@ -211,6 +216,44 @@ def test_multi_master_form_attributes_and_media_versions(self):
         self.assertEqual(_get_form_ids_by_xmlns(linked_master1_build4)[xmlns],
                          _get_form_ids_by_xmlns(linked_master2_build2)[xmlns])
 
+    def test_multi_master_copy_master(self):
```

Comment:
Nice - happy to see tests for this.  The ID versioning is one area I'm still not convinced we've got right, so maybe tests could prove it one way or another.  I know we've discussed it in detail, but IIRC it's not sufficient to check versions from any one upstream app.  For instance:

1. Pull from progenitor app (App A, build 1)
2. Add a new form to App A, not pulled to linked app.
3. Make a copy of App A (called App B)
4. Pull from app B (build 2)
5. Pull from progenitor app (build 3)

In this case, the new form was first seen from App B, so in step 5, when pulling from App A, I think it needs to check build 2 for the new form, but build 1 for the rest.

Reply:
My intent is for that behavior to be handled here: https://github.com/dimagi/commcare-hq/pull/24602/files#diff-87bcd4400226fe34d7fdb28010cfe4b2R178-R182 (and tested here: https://github.com/dimagi/commcare-hq/pull/24602/files#diff-e558ca3134917252d6001108d0939351R123) - let me know what you think, or if we're talking about two different things.

Reply:
Nice, yeah, that looks just like what I was thinking of.  Way ahead of me